lib/rust/ensogl/component/list-view/src/entry/list.rs

```diff
@@ -181,6 +198,17 @@ where E::Model: Default
         }
     }
 
+    /// Set params used in the displayed entries and recreate all displayed entries. The entries
+    /// will use the styles located at the `style_prefix` path in the application's style sheet.
+    pub fn set_entry_params_and_recreate_entries(
+        &self,
+        params: E::Params,
+        style_prefix: style::Path,
+    ) {
+        self.entry_params.replace(params);
+        self.recreate_entries_with_style_prefix(style_prefix);
+    }
```

Comment:
instead of `params: E::Params`, just implement this function for `ListData<E,P>` and use `params::P`. Is this really needed that `E: Entry` for this function to work? Use bounds ONLY when needed. Please check other functions too. Use bounds only if the body of the function requires it.

Reply:
This function calls `recreate_entries_with_style_prefix`, which then calls `Self::update_entry`, which then calls `Entry::update(&self, model: &Entry::Model)`. As such, it needs `E: Entry` to be able to access `Entry::Model`. To me, this seems reasonable, given that the function name contains `recreate_entries`, and (re)creating an entry needs to know the type of `Entry::Model`. Do you have a suggestion on avoiding calling `Entry::update` in this call tree, which would allow us to drop bounds on `E`?